spark-jobs/src/main/scala/za/co/absa/enceladus/common/Constants.scala

```diff
@@ -21,4 +21,16 @@ object Constants {
   final val ReportDateFormat = "yyyy-MM-dd"
   final val InfoVersionColumn = "enceladus_info_version"
   final val EnceladusRecordId = "enceladus_record_id"
+
+  final val SensitiveConfigurationKeys = Set(
```

Comment:
Contrary to the popular opinion, I would stick with this blacklist over a whitelist
It can be tiresome to add whitelist for every settings key we add. Moreover, we control the keys that are used.

I think I also saw sometimes that the key names were matched against some logic that ensured that if it contains e.g. `/password|pwd/i` or similar, it would get redacted, too. That might just be overkill. 

I think we can do with blacklist-filtering, provided there is a test case that check the redaction working properly. In that case, the implementor who would want add a new settings key should be also responsible for making sure that the redaction test case works as well, thus being reminded that the redaction needs to be done where they might have otherwise forgotten.

Reply:
We can use 'redact regex patterns' instead of just a list of keys. We can redact everything that contains 'password', for instance

Reply:
Haha, I was just mentioning that as a thing that we probably do not want to do 🤔. But _Anything's possible™_ , I guess 😆

Reply:
Yeah, if you guys agree, matching against patterns may be an overkill indeed. It might perform poorly as well since we need to match every config key against every pattern 😫. Specifically, in Standardization and Conformance, there are not many keys that need redaction.

I like the solution you have suggested below. It perfectly satisfies requirements we have now and is very elegant.

Reply:
Yeah, I do not really think that the regex way is the key here, anyway. I just thought to mention it to have a more general-solution picture as well.